### ISSUE TITLE
ratbagd.service starts too early during boot, mouse isn't detected as a result

### ISSUE THREAD

**Issue by MrHills-2** [libratbag/libratbag#1667]
**Information**

- `ratbagd` version (`ratbagd --version`):
        0.18
- Distribution:
        Operating System: Arch Linux 
        KDE Plasma Version: 6.2.4
        KDE Frameworks Version: 6.9.0
        Qt Version: 6.8.1
        Kernel Version: 6.12.4-arch1-1 (64-bit)
        Graphics Platform: Wayland
        Processors: 16 × AMD Ryzen 7 7800X3D 8-Core Processor
        Memory: 62.0 GiB of RAM
        Graphics Processor: NVIDIA GeForce RTX 4090/PCIe/SSE2
        Manufacturer: ASUS
- Kernel version (ex. `uname -srmo`): `Linux 6.12.4-arch1-1 x86_64 GNU/Linux`

Device info (if applicable):
```
not relevant
```

**Describe the bug**
I suspect the ratbagd.service starts too early, whenever i try "ratbagctl list" my logitech G502 isn't found. However, if i restart the ratbagd.service with "systemctl restart ratbagd" i can afterwards see my G502 with "ratbagctl list".

**To Reproduce**
Not reliably reproducible.

**Additional context**
Currently fixed by adding

```
[Unit]
After=multi-user.target
After=dev-usb.device
```

Here /etc/systemd/system/ratbagd.service.d/override.conf

**Comment by MrHills-2** [comment]
> ratbagd is started on-demand on D-Bus calls, so tools like Piper start it automatically, it's not supposed to run on boot

I'm a little confused. "systemctl status ratbagd.service" returns me a service that systemd autostarts at boot.

**Comment by BugsyReportsy** [comment]
Same situation on Kubuntu 25.10. From what I can tell, ratbagd.service autostarting doesn't matter, because I still have to manually stop ratbagd.service then reopen Piper or ratbagctl for them to detect my G305, regardless if it's set to enabled or disabled. Oddly enough, this issue didn't seem to occur when I was using it last month, and uninstalling then reinstalling didn't seem to fix anything.

**Comment by PHLAK** [comment]
I've started to have this issue recently as well on Arch using the libratbag-git package from the AUR. The service is enabled but after booting it's not running and my button customizations aren't applied.

```
❭ systemctl status ratbagd.service 
○ ratbagd.service - Daemon to introspect and modify configurable mice
     Loaded: loaded (/usr/lib/systemd/system/ratbagd.service; enabled; preset: disabled)
     Active: inactive (dead)
```

I have to manually start the service for it to start running.

```
❭ systemctl start ratbagd.service 
❭ systemctl status ratbagd.service 
● ratbagd.service - Daemon to introspect and modify configurable mice
     Loaded: loaded (/usr/lib/systemd/system/ratbagd.service; enabled; preset: disabled)
     Active: active (running) since Thu 2026-07-30 17:27:24 MST; 11s ago
 Invocation: 732638cb37af4108a0272abe128fc7d2
   Main PID: 27796 (ratbagd)
      Tasks: 1 (limit: 38013)
     Memory: 2.3M (peak: 2.7M)
        CPU: 56ms
     CGroup: /system.slice/ratbagd.service
             └─27796 /usr/bin/ratbagd
```

The logs don't seem to show anything meaningful.

```
❭ journalctl --boot 0 --unit ratbagd.service
Jul 30 17:27:24 Optimus systemd[1]: Starting Daemon to introspect and modify configurable mice...
Jul 30 17:27:24 Optimus systemd[1]: Started Daemon to introspect and modify configurable mice.
Jul 30 17:48:05 Optimus systemd[1]: ratbagd.service: Deactivated successfully.
```

**Comment by staticssleever668** [comment]
@PHLAK You are not supposed to `systemctl enable ratbagd`, it's a D-Bus-activated service
`systemctl status ratbagd` should show you 'linked' instead of 'enabled' like it does for you

**Comment by staticssleever668** [comment]
> returns me a service that systemd autostarts at boot

Seems to be the same situation

**Comment by PHLAK** [comment]
Thanks @staticssleever668, I disabled the ratbagd service and rebooted my PC and the service shows as `disabled` and `inactive` (not `linked`) until I run `ratbagctl list` or try to launch Piper.

```
❭ systemctl status ratbagd.service 
○ ratbagd.service - Daemon to introspect and modify configurable mice
     Loaded: loaded (/usr/lib/systemd/system/ratbagd.service; disabled; preset: disabled)
     Active: inactive (dead)

❭ ratbagctl list
warbling-capybara:   Logitech G502 X PLUS            
warbling-vole:       Logitech Candy                  
thundering-gerbil:   Logitech G512 RGB MECHANICAL GAMING KEYBOARD

❭ systemctl status ratbagd.service 
● ratbagd.service - Daemon to introspect and modify configurable mice
     Loaded: loaded (/usr/lib/systemd/system/ratbagd.service; disabled; preset: disabled)
     Active: active (running) since Fri 2026-07-31 11:55:46 MST; 4s ago
 Invocation: 6a66f121c6e34813b13fb967ea188093
   Main PID: 3573 (ratbagd)
      Tasks: 1 (limit: 38013)
     Memory: 2.7M (peak: 3.2M)
        CPU: 60ms
     CGroup: /system.slice/ratbagd.service
             └─3573 /usr/bin/ratbagd
```

> [!NOTE]
> On a side note, first time I try to launch piper after a reboot doesn't actually launch Piper but _does_ start the `ratbagd` service. Attempting to launch Piper a second time works as expected.

However, after rebooting my PC and logging in my configured key binds (see screenshot below) do not work. Specifically, the thumb button I have bound to `F13` which I use for push-to-talk in Discord doesn't work.

<img width="1355" height="918" alt="Image" src="https://github.com/user-attachments/assets/379ef2a4-9055-42dc-87ac-38afaabd8a35" />

In order for my push-to-talk key to work I have to run `ratbagctl list` _or_ launch Piper _or_ manually start the `ratbagd` service. Is this the expected behavior? Is there something else I should be doing to start the `ratbagd` service after boot/login? I'm pretty sure this was working as expected up until recently (a few weeks ago maybe).

Happy to open a separate issue for this if needed.

**Comment by staticssleever668** [comment]
> Is there something else I should be doing to start the ratbagd service after boot/login

No, and ratbagd turns off automatically after 20 minutes of not being used.

By the description sounds the same as #1753

**Comment by PHLAK** [comment]
> ratbagd turns off automatically after 20 minutes of not being used

After starting ratbagd once the keybinds work until the next reboot (sometimes days later) so this doesn't seem to be an issue. The only issue is the initial run.

**Comment by staticssleever668** [comment]
@PHLAK I'm not deeply familiar with how Logitech devices work, but could this be related to the "onboard profiles" feature? #1286 mentions that we enabled onboard profiles automatically, but maybe it's a run-time-only option for the mouse (meaning it would reset on PC reboot/mouse reconnection), or we are missing something like a 'commit' that would save it persistently.

**Comment by PHLAK** [comment]
> could this be related to the "onboard profiles" feature?

I'm not too familiar with how this works either but I suppose it's possible. I originally set this mouse up in Windows like a year or two ago and think I remember setting an on-board profile. I don't currently have a Windows PC to check on this though.
